### DISCUSSION TITLE
Bulk import bookmarks from shaarli, wallabag, etc

### DISCUSSION THREAD

**Discussion by OrcVole** [go-shiori/shiori#768] [Feature requests]
Some bookmark collections have taken years to create and considerable time has been spent on organizing them.

For those who like Shiori, it would be lovely to be able to bulk import bookmarks from other bookmarking platforms such as Shaarli, Wallabag, Heimdall or Dashy, for example. This might encourage a lot more users to try Shiori.

Shaarli
https://github.com/shaarli/Shaarli

Wallabag
https://wallabag.org/

Heimdall
https://heimdall.site/

Dashy
https://dashy.to/

**Comment by Monirzadeh** [comment]
If it possible for you please add an exported example file for each service.

**Comment by OrcVole** [comment]
Thank you for asking for the files. I should have supplied these when starting the discussion.

**Shaarli** 
u: demo p: demo
https://demo.shaarli.org/

There are two options for exporting a Shaarli database, with and without prepending the note permalinks with bookmark. Both types are inlcuded.

[bookmarks_all_2
<img width="554" alt="brave_YysAUvjmmD" src="https://github.com/go-shiori/shiori/assets/68787520/80e1d8b5-56f9-4323-8b9e-806edb199a2b">
0231024_130835.zip](https://github.com/go-shiori/shiori/files/13114874/bookmarks_all_20231024_130835.zip)

**Dashy**
This might be cluttered with backup info other than bookmarks, but it is from the live demo
https://demo.dashy.to/
[dashy_conf.zip](https://github.com/go-shiori/shiori/files/13115000/dashy_conf.zip)

Heimdall
The live demo seems to be offline at the moment:
https://www.heimdalldata.com/live-demo/

Wallabag
demo - u: cloudron p: cloudron u: wallabag p: wallabag
https://www.cloudron.io/

[Unread articles.zip](https://github.com/go-shiori/shiori/files/13118789/Unread.articles.zip)

Wallabag allows you to export using a variety of formats, including epub, csv, txt and json

**Comment by fmartingr** [comment]
The import/export system is going to be complicated to support long term.

The main idea now should be to make Shiori be able to export/import to itself using our own file format, the system should be refactored to use an interface and a way to register other importers/exporters.

Then the community may help with that, but us providing ways to export/import from a bunch of services/foss projects is just not feasible at this point.

**Reply by Monirzadeh** [reply]
i thinking about one functionality that import an specific format.
community write script to change any service export to that specific format.
user get backup from any other service --> covert to format that shiori understand that (with community)--> import in shiori UI

currently i cant show progress of importing possess as i want.

**Reply by fmartingr** [reply]
If you have given this any though, please open an issue to discuss how the import/export system should work and be integrated before you start implementing it, just so that everything is correctly aligned.
